cdp_scrapers/legistar_utils.py

```diff
@@ -109,3 +184,805 @@ def get_legistar_events_for_timespan(
+
+    @property
+    def is_legistar_compatible(self) -> bool:
+        """
+        Check that Legistar API recognizes client name
+
+        Returns
+        -------
+        bool
+            True if client_name is a valid Legistar client name
+        """
+        # simplest check, if the GET request works, it is a legistar place
+        try:
+            resp = urlopen(f"http://webapi.legistar.com/v1/{self.client_name}/bodies")
+            return resp.status == 200
+        except URLError or HTTPError:
+            return False
+
+    def check_for_cdp_min_ingestion(self, check_days: int = 7) -> bool:
+        """
+        Test if can obtain at least one minimally defined EventIngestionModel
+
+        Parameters
+        ----------
+        check_days : int, default=7
+            Test duration is the past check_days days from now
+
+        Returns
+        -------
+        bool
+            True if got at least one minimally defined EventIngestionModel
+        """
+        # no point wasting time if the client isn't on legistar at all
+        if not self.is_legistar_compatible:
+            return False
+
+        now = datetime.utcnow()
+        days = range(check_days)
+
+        for d in days:
+            begin = now - timedelta(days=d + 1)
+            end = now - timedelta(days=d)
+            log.debug(
+                "Testing for minimal information "
+                f"from {begin.isoformat()} to {end.isoformat()}"
+            )
+
+            # ev: EventIngestionModel
+            for cdp_ev in self.get_events(begin=begin, end=end):
+                try:
+                    if (
+                        len(cdp_ev.body.name) > 0
+                        and cdp_ev.sessions[0].session_datetime is not None
+                        and len(cdp_ev.sessions[0].video_uri) > 0
+                    ):
+                        session_time = cdp_ev.sessions[0].session_datetime
+                        log.debug(
+                            f"Got minimal EventIngestionModel for {self.client_name}: "
+                            f"body={cdp_ev.body.name}, "
+                            f"session={session_time.isoformat()}, "
+                            f"video={cdp_ev.sessions[0].video_uri}"
+                        )
+                        return True
+
+                # catch None or empty list
+                except TypeError or IndexError:
+                    pass
+
+        log.debug(
+            f"Failed to get minimal EventIngestionModel for {self.client_name} "
+            f"in the past {check_days} days from {now.isoformat()}"
+        )
+        # no event in check_days had enough for minimal ingestion model item
+        return False
+
+    def get_events(
+        self,
+        # for the past 2 days
+        begin: Optional[datetime] = datetime.utcnow() - timedelta(days=2),
+        end: Optional[datetime] = datetime.utcnow(),
+    ) -> List[EventIngestionModel]:
```

Comment:
Nit: can we set these `begin` and `end` to `Optional[datetime] = None`.

`Optional` typically means the default value is `None` and specifically in this case, by setting the default value to a datetime you are taking the datetime when the module is imported. I.e. say I have a long running script, when I import this module it will set the default values of the parameters to the current date but when the long running script actually hits the function call it uses now, day-old datetimes.

Instead, set these to `None` and then:
```python
if begin is None:
    begin = datetime.utcnow()....
```